### ISSUE TITLE
CSV files and validation sets

### ISSUE THREAD

**Issue by romanceyu** [ArnabKumarRoy02/ResEmoteNet#36]
Hello, my friend. I still do not fully understand how you generate the CSV files for the four major datasets and how you handle the file name splitting operations. Could you also explain how you addressed the issue of missing validation sets in the dataset? In other words, how was your validation set generated? Is it true that a portion of the augmented training set is separated out to form the validation set? And is there a potential risk of data leakage or the validation set being included in the training set? I mean no offense, I am sincerely asking for your guidance.

**Comment by ArnabKumarRoy02** [comment]
Hi @romanceyu, 

For the datasets with no validation set (like RAF-DB or AffectNet), we took 20% of the augmented training set and used that as the validation set. The files used in the validation set were removed from the training set to overcome the risk of data leakage making both the sets unique.

For the creation of the CSV file, use the [data_csv.py](https://github.com/ArnabKumarRoy02/ResEmoteNet/blob/main/data_preprocessing/data_csv.py) file. Just make sure to give the path of the different sets.

## Working of the above script:
Lets say you have the training set as:
```bash
train/
        train_1_happy.png
        train_2_sad.png
        train_3_happy.png
        ...
```

Using the above script will create the CSV file for training set as:
```bash
train_1_happy.png, 0
train_2_sad.png, 2
train_3_happy.png, 0
```

Make sure of the emotion names in the filename. Update the emotions accordingly in the script. For some dataset: "anger" can be represented as "angry", "sad" can be "sadness" and so on.

Hope this explanation clarifies the process and facilitates your understanding.
